sphinx/themes/basic/static/searchtools.js

```diff
@@ -219,6 +219,18 @@ const Search = {
     (document.body.appendChild(document.createElement("script")).src = url),
 
   setIndex: (index) => {
+    const stack = [index];
+    while (stack.length) {
+      const value = stack.pop();
+      if (!(value instanceof Object)) continue;
```

Comment:
Here we immediately discard any non-object items popped from the `stack` array.

Perhaps it would be cheaper/faster not to place those items onto the stack in the first place?